src/lib/tuning/index/index_choice.cpp

```diff
@@ -0,0 +1,30 @@
+#include "index_choice.hpp"
+
+#include <memory>
+
+#include "tuning/index/index_operation.hpp"
+
+namespace opossum {
+
+float IndexChoice::desirability() const { return saved_work; }
+
+float IndexChoice::cost() const { return memory_cost; }
+
+float IndexChoice::confidence() const { return 1.0f; }
+
+bool IndexChoice::is_currently_chosen() const { return exists; }
```

Comment:
I seem to remember that you directly accessed `exists` somewhere as well. Either use the method (and make the member protected) OR the member. This is even more important when they have different names.

Reply:
Please review our comment on line 31 in index_choice.hpp.

We wanted to keep the public members, as IndexChoice plays a dual role as a data structure created and processed by the IndexEvaluator, which uses it as a data class and thus its index specific public members. On the other hand it acts as an abstract TuningChoice and must expose its information compatible to this interface.
We think it would be more confusing to let the index-specific code in IndexEvaluator use the abstract names of TuningChoice.

We could however write a separate set of accessors that are specific to the IndexChoice, if you definitely do not want access to public members.